src/sssom/parsers.py

```diff
@@ -566,118 +568,104 @@ def from_obographs(
-                        for ec1 in equivalents["nodeIds"]:
-                            for ec2 in equivalents["nodeIds"]:
-                                if ec1 != ec2:
-                                    mdict = {}
-                                    mdict[SUBJECT_ID] = safe_compress(ec1, converter)
-                                    mdict[OBJECT_ID] = safe_compress(ec2, converter)
-                                    mdict[PREDICATE_ID] = safe_compress(
-                                        OWL_EQUIV_CLASS_URI, converter
-                                    )
-                                    mdict[MAPPING_JUSTIFICATION] = MAPPING_JUSTIFICATION_UNSPECIFIED
-                                    mdict[SUBJECT_LABEL] = (
-                                        labels[ec1] if ec1 in labels.keys() else ""
-                                    )
-                                    mdict[OBJECT_LABEL] = (
-                                        labels[ec2] if ec2 in labels.keys() else ""
-                                    )
-                                    _add_valid_mapping_to_list(mdict, mlist)
-    else:
+    graphs = jsondoc.get("graphs")
+    if not graphs:
         raise Exception("No graphs element in obographs file, wrong format?")
 
+    #: A dictionary of node URIs to node labels
+    labels: Mapping[str, str] = {
+        node["id"]: node.get("lbl")
+        for graph in graphs
+        for node in graph.get("nodes", [])
+        if node.get("lbl")
+    }
+
+    for graph in graphs:
+        for node in graph.get("nodes", []):
+            meta = node.get("meta")
+            if not meta:
+                continue
+
+            node_uri = node["id"]
+            if DBXREF_URI in mapping_predicates:
+                for xref in meta.get("xrefs", []):
+                    mdict = _make_mdict(node_uri, DBXREF_URI, xref["val"], converter, labels)
+                    _add_valid_mapping_to_list(mdict, mlist)
+
+            for value in meta.get("basicPropertyValues", []):
+                predicate_uri = value["pred"]
+                if predicate_uri not in mapping_predicates:
+                    continue
+                mdict = _make_mdict(node_uri, predicate_uri, value["val"], converter, labels)
+                _add_valid_mapping_to_list(mdict, mlist)
+
+        for edge in graph.get("edges", []):
+            predicate_uri = _get_obographs_predicate_id(edge["pred"])
+            if predicate_uri not in mapping_predicates:
+                continue
+            mdict = _make_mdict(edge["sub"], predicate_uri, edge["obj"], converter, labels)
+            _add_valid_mapping_to_list(mdict, mlist)
+
+        if OWL_EQUIV_CLASS_URI in mapping_predicates:
+            for equivalents in graph.get("equivalentNodesSets", []):
+                node_ids = equivalents.get("nodeIds")
+                if not node_ids:
+                    continue
+                for subject_uri, object_uri in itt.product(node_ids, repeat=2):
```

Comment:
What is itt?

Reply:
itertools. Importing as `itt` is very common

Reply:
Thanks